src/ORM/Association/HasMany.php

```diff
@@ -117,6 +165,10 @@ public function saveAssociated(EntityInterface $entity, array $options = [])
                 $targetEntity->set($properties, ['guard' => false]);
             }
 
+            if ($this->_saveStrategy === self::SAVE_REPLACE) {
+                $targetEntity->isNew(true);
```

Comment:
Why are we deleting entities just to reinsert them? That can make created timestamps incorrect. Is there a way we can do diffs?